track_analyzing/track_analyzer/cmd_table.cpp

```diff
@@ -285,36 +312,64 @@ class MatchedTrackPointToMoveType final
   {
     if (container.IsExist(CITY_ROADS_FILE_TAG))
       LoadCityRoads(container.GetFileName(), container.GetReader(CITY_ROADS_FILE_TAG), m_cityRoads);
+
+    if (container.IsExist(MAXSPEED_FILE_TAG))
+      LoadMaxspeeds(container.GetReader(MAXSPEED_FILE_TAG), m_maxspeeds);
   }
 
   MoveType GetMoveType(MatchedTrackPoint const & point)
   {
     auto const & dataPoint = point.GetDataPoint();
-    return MoveType(GetRoadInfo(point.GetSegment().GetFeatureId()),
+    return MoveType(GetRoadInfo(point.GetSegment()),
                     static_cast<traffic::SpeedGroup>(dataPoint.m_traffic),
-                    DayTimeToBool(GetDayTime(dataPoint.m_timestamp, dataPoint.m_latLon.lat, dataPoint.m_latLon.lon)));
+                    dataPoint);
   }
 
 private:
-  RoadInfo GetRoadInfo(uint32_t featureId)
+  RoadInfo GetRoadInfo(Segment const & segment)
   {
+    auto const featureId = segment.GetFeatureId();
     if (featureId == m_prevFeatureId)
       return m_prevRoadInfo;
 
     FeatureType feature;
     m_featuresVector.GetVector().GetByIndex(featureId, feature);
 
+    auto const maxspeed = m_maxspeeds.GetMaxspeed(featureId);
+    auto maxspeedValue = kInvalidSpeed;
+    if (maxspeed.IsValid())
+    {
+      if (maxspeed.IsBidirectional())
+        maxspeedValue = segment.IsForward() ? maxspeed.GetForward() : maxspeed.GetBackward();
+      else
+        maxspeedValue = maxspeed.GetForward();
+
+      if (maxspeed.GetUnits() == measurement_utils::Units::Imperial)
+        maxspeedValue = measurement_utils::MphToKmph(maxspeedValue);
```

Comment:
It's better to use `Maxspeed::GetSpeedKmPH(bool forward) const` instead of the code above.